### PR TITLE
[Feature][doc][Connector-V2][Common] Add Common connector documentation

### PR DESCRIPTION
**Pull request by zhilinli123**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by zhilinli123** [comment]
PTAL: @liugddx @EricJoy2048

File: docs/en/connector-v2/sink/common-options.md [hunk 1]
```diff
@@ -2,24 +2,16 @@
...
+|-------------------|--------|----------|---------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| source_table_name | String | No       | -       | When `source_table_name` is not specified, the current plug-in processes the data set `dataset` output by the previous plugin in the configuration file <br/> When `source_table_name` is specified, the current plug-in is processing the data set corresponding to this parameter. |
+| parallelism       | Int    | No       | -       | When `parallelism` is not specified, the `parallelism` in env is used by default. <br/>When parallelism is specified, it will override the parallelism in env.                                                                                                                       |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/common-options.md]: At present, we do not support setting the parallelism for Sink separately. I think we can remove this parameter

Comment (zhilinli123) [reply docs/en/connector-v2/sink/common-options.md]: Done

File: docs/en/connector-v2/sink/common-options.md [hunk 2]
```diff
@@ -2,24 +2,15 @@
...
-|-------------------|--------|----------|---------------|
-| source_table_name | string | no       | -             |
-| parallelism       | int    | no       | -             |
+|       Name        |  Type  | Required | Default |                                                                                                                                     Description                                                                                                                                      |
+|-------------------|--------|----------|---------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| source_table_name | String | No       | -       | When `source_table_name` is not specified, the current plug-in processes the data set `dataset` output by the previous plugin in the configuration file <br/> When `source_table_name` is specified, the current plug-in is processing the data set corresponding to this parameter. |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/common-options.md:7]: Some important caveat should be added: `source_table_name` and `result_table_name` are only used to represent the upstream and downstream relationships of the connector. They should not be treated as in-memory datasets. No data is stored in them.

File: docs/en/connector-v2/source/common-options.md [hunk 3]
```diff
@@ -2,32 +2,76 @@
...
+| result_table_name | String | No       | -       | When `result_table_name` is not specified, the data processed by this plugin will not be registered as a data set `(dataStream/dataset)` that can be directly accessed by other plugins, or called a temporary table `(table)` <br/>When `result_table_name` is specified, the data processed by this plugin will be registered as a data set `(dataStream/dataset)` that can be directly accessed by other plugins, or called a temporary table `(table)` . The data set `(dataStream/dataset)` registered here can be directly accessed by other plugins by specifying `source_table_name` . |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/common-options.md:7]: same as above.

File: docs/en/transform-v2/common-options.md [hunk 4]
```diff
@@ -1,23 +1,65 @@
...
+|-------------------|--------|----------|---------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| result_table_name | String | No       | -       | When `source_table_name` is not specified, the current plug-in processes the data set `(dataset)` output by the previous plug-in in the configuration file; <br/>When `source_table_name` is specified, the current plugin is processing the data set corresponding to this parameter.                                                                                                                                                                                                                                                |
```
Comment (EricJoy2048) [inline docs/en/transform-v2/common-options.md:7]: same as above.

**Comment by zhilinli123** [comment]
PTAL : @Carl-Zhou-CN @hailin0 @EricJoy2048

File: docs/en/transform-v2/common-options.md [hunk 5]
```diff
@@ -1,23 +1,65 @@
...
+            c_int = int
+          }
+        }
+      }
+    }
+  }
+}
 
-When `result_table_name` is specified, the data processed by this plugin will be registered as a data set `(dataset)` that can be directly accessed by other plugins, or called a temporary table `(table)` . The dataset registered here can be directly accessed by other plugins by specifying `source_table_name` .
+transform {
```
Comment (EricJoy2048) [inline docs/en/transform-v2/common-options.md:45]: Need add a example for Multiple Tables.

Comment (zhilinli123) [reply docs/en/transform-v2/common-options.md:45]: > Need add a example for Multiple Tables.

@EricJoy2048 Do you have a specific reference case let me add

**Review by EricJoy2048 — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**